---
title: "05-Hisat"
output: github_document
date: "2023-12-12"
---

Code for RNAseq of 2022 SeaStar experiment

```{r setup, include=FALSE}
library(knitr)
library(tidyverse)
library(DT)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,         # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center" # Align plots to the center
)
```

# Sample info

```{r, eval=TRUE}
mdata <- read.delim("../data/summer2022_samples_sequenced.csv", sep = ",")
head(mdata)
```


```{r, eval=TRUE}
#simplified metadata
smdata <- select(mdata, coelom_ID, star_ID, star_size, treatment_grp, experiment_day, disease_sign)

smdata

```


```{r, eval=FALSE}
datatable(smdata, options = list(scrollX = TRUE, scrollY = "400px", scrollCollapse = TRUE, paging = FALSE))

```

# Trimmed Reads

On Gannet
https://gannet.fish.washington.edu/gcrandall/bu-mox/20231206_PSC2022_trimming/

```{bash}
wget -r \
--no-directories --no-parent \
-P ../data \
-A "*.fq.gz" https://gannet.fish.washington.edu/gcrandall/bu-mox/20231206_PSC2022_trimming/
```



# HiSat Indexing of Genome





# genome

```{bash}
cd ../data
/home/shared/datasets download genome accession GCA_032158295.1 --include gff3,rna,cds,protein,genome,seq-report
```

then moved around



#genome prep

Just fasta... 01  - will not initially use as does not have splice site info. Will keep if ever want to do comparison.

```{r, engine='bash'}
/home/shared/hisat2-2.2.1/hisat2-build \
../data/GCA_032158295.1_ASM3215829v1_genomic.fna \
../analyses/05-hisat/GCA_032158295.1_01 \
-p 40 \
> ../analyses/05-hisat/GCA_032158295.1_01.txt

```


# genome feature files

https://datadryad.org/stash/dataset/doi:10.5061/dryad.51c59zwfd


```{r, engine='bash', eval=TRUE}
head ../data/aug*
ls ../data/aug*
```


# extract exons and splice sites

```{r, engine='bash'}
/home/shared/hisat2-2.2.1/hisat2_extract_exons.py \
../data/augustus.hints.gtf \
> ../analyses/05-hisat/exon.tab
```

```{r, engine='bash', eval=TRUE}
head ../analyses/05-hisat/exon.tab
```



```{r, engine='bash'}
/home/shared/hisat2-2.2.1/hisat2_extract_splice_sites.py \
../data/augustus.hints.gtf \
> ../analyses/05-hisat/splice_sites.tab

```


```{r, engine='bash', eval=TRUE}
head ../analyses/05-hisat/splice_sites.tab
```



# Indexing with splice sites

Genome indexing- `02` suffix; has exon and splice site tbas incorporated.


```{r, engine='bash'}
/home/shared/hisat2-2.2.1/hisat2-build \
../data/GCA_032158295.1_ASM3215829v1_genomic.fna \
../analyses/05-hisat/GCA_032158295.1_02 \
--exon ../analyses/05-hisat/exon.tab \
--ss ../analyses/05-hisat/splice_sites.tab \
-p 40 \
../data/augustus.hints.gtf \
2> ../analyses/05-hisat/GCA_032158295.1_02.txt
```






# Alignment


```{r, engine='bash'}

# Loop through all  files
for file in ../data/*_R1*fq.gz; do
    # Remove the  part to get the base name
    base=$(basename "$file" _R1_001.fastq.gz.fastp-trim.20231206.fq.gz)
    # Construct the names of the pair of files
    file1=${base}_R1_001.fastq.gz.fastp-trim.20231206.fq.gz
    file2=${base}_R2_001.fastq.gz.fastp-trim.20231206.fq.gz
    # Run the hisat2 command
    /home/shared/hisat2-2.2.1/hisat2 \
    -x ../analyses/05-hisat/GCA_032158295.1_02 \
    -p 20 \
    -1 ../data/$file1 \
    -2 ../data/$file2 \
    -S ../analyses/05-hisat/${base}.sam \
    2> ../analyses/05-hisat/${base}_hisat2_stats.txt
done


```

# Converting sam to sorted bams

```{r, engine='bash'}
for samfile in ../analyses/05-hisat/*.sam; do
  bamfile="${samfile%.sam}.bam"
  sorted_bamfile="${samfile%.sam}.sorted.bam"
  
  # Convert SAM to BAM
  /home/shared/samtools-1.12/samtools view -bS -@ 20 "$samfile" > "$bamfile"
  
  # Sort BAM
  /home/shared/samtools-1.12/samtools sort -@ 20 "$bamfile" -o "$sorted_bamfile"
  
  # Index sorted BAM
  /home/shared/samtools-1.12/samtools index -@ 20 "$sorted_bamfile"
done
```



# Stringtie



```{bash}
cd ../data
curl -O https://gannet.fish.washington.edu/seashell/snaps/interproscan.gff3
```
```{r, engine='bash', eval=TRUE}
head ../data/interproscan.gff3 
```



```{r, engine='bash'}
find ../analyses/05-hisat/*sorted.bam \
| xargs basename -s .sorted.bam | xargs -I{} \
/home/shared/stringtie-2.2.1.Linux_x86_64/stringtie \
-p 20 \
-o ../analyses/05-hisat/{}.gtf \
../analyses/05-hisat/{}.sorted.bam
```


```{r, engine='bash', eval=TRUE}
head ../analyses/05-hisat/*.gtf | head -25

```

