---
title: "08-deseq2-controlvarmdrop"
output: html_document
date: "2023-12-28"
---
Rmd to use `DESeq2` to get differentially expressed genes lists from summmer 2022 samples. Experiment was comparing the response to exposure of SSWD between small (juvenile) and large (adult) _Pycnopodia helianthoides_ and compared to control stars. 

Sample info: 
https://github.com/grace-ac/project-pycno-sizeclass-2022/blob/main/data/summer2022_metadata.csv 
Load packages:    
installed 2023-12-26 
```{r}
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("DESeq2")
```


```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
library(vegan)
library(broom)
```
installed 2024-01-02
```{r}
#install.packages("vegan")
#install.packages("broom")
```


Read in count matrix created on Raven using: https://github.com/grace-ac/project-pycno-sizeclass-2022/blob/main/code/07-kallisto-summer2022-phelgenome.Rmd 

```{r}
countmatrix <- read.delim("../analyses/07-kallisto-phelgenome/kallisto_20231204.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

Round integers up to hole numbers for further analysis:
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

# For SICB: Compare Control vs Arm Drop (only looking at libraries from Day 8-15) 


coelom_ID	star_size	experiment_day	disease_sign	star_ID
PSC.0149	large	                8	first_arm_drop	28
PSC.0174	large	                10	first_arm_drop	30
PSC.0177	small	                11	healthy	34
PSC.0186	large	                11	healthy	15
PSC.0187	small	                11	first_arm_drop	57
PSC.0188	small	                11	first_arm_drop	38
PSC.0190	large               	11	first_arm_drop	10
PSC.0202	large               	12	healthy	4
PSC.0203	large               	12	healthy	31
PSC.0209	large               	13	healthy	9
PSC.0219	small               	14	healthy	63
PSC.0228	small	                14	first_arm_drop	61
PSC.0230	small	                15	healthy	65
PSC.0231	large	                15	first_arm_drop	13

Subset the above libraries:
```{r}
#control (con) vs armdrop (ad)
convad <- select(countmatrix, "PSC.0149", "PSC.0174", "PSC.0177", "PSC.0186", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0202", "PSC.0203", "PSC.0209", "PSC.0219", "PSC.0228", "PSC.0230", "PSC.0231")
head(convad)
```
Get DEGs comparing the two groups - control vs first arm drop. 
```{r}
colData <- data.frame(condition=factor(c("exposed", "exposed", "healthy", "healthy", "exposed", "exposed", "exposed", "healthy", "healthy", "healthy", "healthy", "exposed", "healthy", "exposed")), 
                             type=factor(rep("paired-end", 14)))
rownames(colData) <- colnames(convad)
dds <- DESeqDataSetFromMatrix(countData = convad,
                                     colData = colData, 
                                     design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```
```{r}
head(res)
```
From [Bioconductor `DESeq2` Vignette](http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html):     
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```
```{r}
plot <- plotPCA(vsd, intgroup=c("condition"))
nudge <- position_nudge(y = 4)
plot + geom_text(aes(label = name), position = nudge)
```

```{r}
plotrld <- plotPCA(rld, intgroup=c("condition", "type"))
nudge <- position_nudge(y = 4)
plotrld + geom_text(aes(label = name), position = nudge)
```


Create a permanova plot to get statistical significance if any:
transpose the vsd data so that the genes are the columns and the rows (n=14) are the libraries 
```{r}
vsd.transpose <- t(assay(vsd))
head(vsd.transpose)
```

vsd.transpose is currently a matrix, but adonis2 needs it to be a data frame. Make it into a dataframe:
```{r}
vsd.transpose.df=as.data.frame(vsd.transpose)
head(vsd.transpose.df)
```
make rownames into a column called LibraryID:
```{r}
library(tibble)
vsd.transpose.df <- tibble::rownames_to_column(vsd.transpose.df, "LibraryID")
head(vsd.transpose.df)
```

Add columns to dataframe to explain the libraries:
```{r}
vsd.transpose.df$condition <- c("exposed", "exposed", "healthy", "healthy", "exposed", "exposed", "exposed", "healthy", "healthy", "healthy", "healthy", "exposed", "healthy", "exposed")
head(vsd.transpose.df)
```



```{r}
permanova.convad <- adonis2(scale(vsd.transpose.df[c(2:26582)]) %>%
                       replace(is.na(.), 0) ~ condition, data = vsd.transpose.df, method = "eu")  
permanova.convad #Significant influence of all factors
```
```{r}
#broom::tidy(permanova.convad) %>%
#  write.csv("../analyses/08-deseq2/7controlV7armdrop_Permanova_results.csv", quote = FALSE) #Save PERMANOVA output
```
write out 2024-01-02.

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
```{r}
control_v_armdrop_fig <- res
# The main plot
plot(control_v_armdrop_fig$baseMean, control_v_armdrop_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Control vs Exposed (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
control_v_armdrop_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(control_v_armdrop_fig.sig$baseMean, control_v_armdrop_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```
Write out list of those 7117 DEGs:
Wrote out 2023-12-28. 
```{r}
#write.table(control_v_armdrop_fig.sig, "../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```

# `join` deg lists with library counts and blast output
load the `../analyses/08-deseq2/DEGlist_control_v_armdrop.tab`
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7.tab")
head(deglist)
```

`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
convadrownames <- rownames_to_column(convad, var = "gene")
head(convadrownames)
```
```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```
```{r}
deglistcount <- left_join(deglist, convadrownames, by = "gene")
head(deglistcount)
```
write out deglist count:
```{r}
#write.table(deglistcount, "../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7_genecounts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
write out 20240101. 

read in blast output
```{r}
blastout <- read.delim("../analyses/10-BLAST/summer2022-uniprot_blastx_sep.tab", header = FALSE)
head(blastout)
```

rename the first column "gene":
```{r}
cols <- c("gene", "V2", "uniprotID", "geneID", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
colnames(blastout) <- cols
head(blastout)
```
`join` with deglistcount by "gene" column:
```{r}
deglistcountblast <- left_join(deglistcount, blastout, by = "gene")
head(deglistcountblast)
```

Write out the file:
```{r}
#write.table(deglistcountblast, "../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7_genecounts_blast.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231228

## MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```
Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data:    
Pull the columns out in order where controls are first:     
```{r}
degheatmap <- select(degheatmap, "PSC.0177", "PSC.0186", "PSC.0202", "PSC.0203", "PSC.0209", "PSC.0219", "PSC.0230", "PSC.0149", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(degheatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```
Should be 7117 rows

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with healthy or firstarmdrop:
```{r}
sample_col <- data.frame(treatment = c("control", "control", "control", "control", "control", "control", "control", "exposed", "exposed", "exposed", "exposed", "exposed", "exposed", "exposed"))
row.names(sample_col) <-colnames(nozerodegheatmap)
sample_col
```
add annotation for star size:
```{r}
sample_col$star.size <- c("small", "large", "large", "large", "large", "small", "small", "large", "large", "small", "small", "large", "small", "large")
sample_col
```

specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "cyan3", exposed = "coral3")
  #star.size = c(large = "#AF2CEC", small = "#2C75EC")
)
```

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```

## Now see if size/age of star has an impact on this:
Code based on the Multi-factor design section of the manual:        http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs 

```{r}
colData <- data.frame(condition=factor(c("control", "control", "control", "control", "control", "control", "control", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop")), 
                             type=factor(rep("paired-end", 14)),
                             size=factor(c("small", "large", "large", "large", "large", "small", "small", "large", "large", "small", "small", "large", "small", "large")))
rownames(colData) <- colnames(convad)
dds <- DESeqDataSetFromMatrix(countData = convad,
                                     colData = colData, 
                                     design = ~ condition + size)
```
Check levels of size and condition
```{r}
levels(dds$size)
```
```{r}
levels(dds$condition)
```
The following will pull the results from `size` (large or small) because that is our variable of interest. This tells us how size/age contributes to the infection DEGs
```{r}
design(dds) <- formula(~ condition + size)
dds <- DESeq(dds)
```
Access results:
```{r}
res <- results(dds)
head(res)
```

```{r}
summary(res)
```
make a PCA:
```{r}
vsd <- vst(dds, blind=FALSE)
rld <- rlog(dds, blind=FALSE)
head(assay(vsd), 3)
```

```{r}
plot <- plotPCA(vsd, intgroup=c("condition", "size"))
nudge <- position_nudge(y = 4)
plot + geom_text(aes(label = name), position = nudge)
```

run PERMANOVA:
transpose the vsd data so that the genes are the columns and the rows (n=14) are the libraries 
```{r}
vsd.transpose <- t(assay(vsd))
head(vsd.transpose)
```

vsd.transpose is currently a matrix, but adonis2 needs it to be a data frame. Make it into a dataframe:
```{r}
vsd.transpose.df=as.data.frame(vsd.transpose)
head(vsd.transpose.df)
```

make rownames into a column called LibraryID:
```{r}
library(tibble)
vsd.transpose.df <- tibble::rownames_to_column(vsd.transpose.df, "LibraryID")
head(vsd.transpose.df)
```

Add columns to dataframe to explain the libraries:
```{r}
vsd.transpose.df$condition <- c("exposed", "exposed", "healthy", "healthy", "exposed", "exposed", "exposed", "healthy", "healthy", "healthy", "healthy", "exposed", "healthy", "exposed")
head(vsd.transpose.df)
```
```{r}
permanova.convad <- adonis2(scale(vsd.transpose.df[c(2:26582)]) %>%
                       replace(is.na(.), 0) ~ condition, data = vsd.transpose.df, method = "eu")  
permanova.convad #Significant influence of all factors
```
```{r}
#broom::tidy(permanova.convad) %>%
#  write.csv("../analyses/08-deseq2/7controlV7armdrop_withAge_Permanova_results.csv", quote = FALSE) #Save PERMANOVA output
```
write out 20240111.
```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
one??

```{r}
sizeconvad <- res
# The main plot
plot(sizeconvad$baseMean, sizeconvad$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main= "Control vs Arm Drop Stars Taking Size into Accoun  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
sizeconvad.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(sizeconvad.sig$baseMean, sizeconvad.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```


```{r}
#write.table(sizeconvad.sig, "../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7_contrastsize.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out table 20231228. Commented out code

## `join` deg lists with library counts
load the ../analyses/08-deseq2/DEGlist_control_v_armdrop_andsize.tab
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7_contrastsize.tab")
head(deglist)
```

`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
convadrownames <- rownames_to_column(convad, var = "gene")
head(convadrownames)
```

```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, convadrownames, by = "gene")
head(deglistcount)
```
`join` with blastoutput
```{r}
blastout <- read.delim("../analyses/10-BLAST/summer2022-uniprot_blastx_sep.tab", header = FALSE)
head(blastout)
```
rename column 1 gene:
```{r}
cols <- c("gene", "V2", "uniprotID", "geneID", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
colnames(blastout) <- cols
head(blastout)
```

`join` with deglistcount
```{r}
deglistcountblast <- left_join(deglist, blastout, by = "gene")
head(deglistcountblast)
```

write out file:
write out the blast annotated file:
```{r}
#write.table(deglistcountblast, "../analyses/08-deseq2/DEGlist_control_v_armdrop_7x7_contrastsize_genecounts_blast.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231228

# MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data in order of star size:
```{r}
degheatmap <- select(degheatmap, "PSC.0177", "PSC.0186", "PSC.0202", "PSC.0203", "PSC.0209", "PSC.0219", "PSC.0230", "PSC.0149", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(degheatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```










