---
title: "08-deseq2.Rmd"
output: html_document
date: "2023-12-14"
---
Rmd to use `DESeq2` to get differentially expressed genes lists from summmer 2022 samples. Experiment was comparing the response to exposure of SSWD between small (juvenile) and large (adult) _Pycnopodia helianthoides_ and compared to control stars. 

Sample info: 
https://github.com/grace-ac/project-pycno-sizeclass-2022/blob/main/data/summer2022_metadata.csv 
Load packages:    
installed 2023-12-26 
```{r}
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("DESeq2")
```


```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Read in count matrix created on Raven using: https://github.com/grace-ac/project-pycno-sizeclass-2022/blob/main/code/07-kallisto-summer2022-phelgenome.Rmd 

```{r}
countmatrix <- read.delim("../analyses/07-kallisto-phelgenome/kallisto_20231204.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

Round integers up to hole numbers for further analysis:
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

# Compare: Exposed Large Adult _Pycnopodia helianthoides_ (n=4) to Control Large Adult _Pycnopodia helianthoides_ (n=4)

Samples:
Exposed: PSC.0190; PSC.0206; PSC.0217; PSC.0231     
Control: PSC.0186; PSC.0202; PSC.0203; PSC.0209    

Pull out those samples:
```{r}
#levlc is large exposed vs large control
levlc <- select(countmatrix, "PSC.0190", "PSC.0206", "PSC.0217", "PSC.0231", "PSC.0186", "PSC.0202", "PSC.0203", "PSC.0209")
head(levlc)
```

Get DEGs based on exposure to SSWD / disease sign (all exposed adults had at least 1 dropped arm and some had twisting)

```{r}
colData <- data.frame(condition=factor(c("exposed", "exposed", "exposed", "exposed", "control", "control", "control", "control")), 
                             type=factor(rep("paired-end", 8)))
rownames(colData) <- colnames(levlc)
dds <- DESeqDataSetFromMatrix(countData = levlc,
                                     colData = colData, 
                                     design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
[1] 4617    6

```{r}
adult_exposure_fig <- res
# The main plot
plot(adult_exposure_fig$baseMean, adult_exposure_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="SSWD Exposure Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
adult_exposure_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(adult_exposure_fig.sig$baseMean, adult_exposure_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

# Compare: Exposed Small Juvenile _Pycnopodia helianthoides_ (n=4) to Control Small Juvenile _Pycnopodia helianthoides_ (n=4)

Samples:    
Exposed: PSC.0187; PSC.0188; PSC.0198; PSC.0228     
Control: PSC.0141; PSC.0177; PSC.0219; PSC.0230

Pull out those samples:
```{r}
#sevsc is small exposed vs small control
sevsc <- select(countmatrix, "PSC.0187", "PSC.0188", "PSC.0198", "PSC.0228", "PSC.0141", "PSC.0177", "PSC.0219", "PSC.0230")
head(sevsc)
```

```{r}
colData <- data.frame(condition=factor(c("exposed", "exposed", "exposed", "exposed", "control", "control", "control", "control")), 
                             type=factor(rep("paired-end", 8)))
rownames(colData) <- colnames(sevsc)
dds <- DESeqDataSetFromMatrix(countData = sevsc,
                                     colData = colData, 
                                     design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
[1] 6125    6

```{r}
juvenile_exposure_fig <- res
# The main plot
plot(juvenile_exposure_fig$baseMean, juvenile_exposure_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="SSWD Exposure Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
juvenile_exposure_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(juvenile_exposure_fig.sig$baseMean, juvenile_exposure_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```


# Compare the libraries of stars sampled at arm drop, taking into account age/size, and days since exposure (experiment day)

| Library  | Star_Size | Experiment Day (Days since Exposure to SSWD) |
|----------|-----------|----------------------------------------------|
| PSC 0149 |     large | 8                                            |
| PSC 0150 |     large | 8                                            |
| PSC 0174 |     large | 10                                           |
| PSC 0187 |     small | 11                                           |
| PSC 0188 |     small | 11                                           |
| PSC 0190 |     large | 11                                           |
| PSC 0228 |     small | 14                                           |
| PSC 0231 |     large | 15                                           |


```{r}
armdrop <- select(countmatrix, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(armdrop)
```

Create a dataframe to get DEGs based on difference in age/size and taking experimental day into account: 
```{r}
colData <- data.frame(condition=factor(c("large", "large", "large", "small", "small", "large", "small", "large")), 
                             type=factor(rep("paired-end", 8)),
                      day=factor(c("day8", "day8", "day10", "day11", "day11", "day11", "day14", "day15")))
rownames(colData) <- colnames(armdrop)
dds <- DESeqDataSetFromMatrix(countData = armdrop,
                                     colData = colData, 
                                     design = ~ condition + day)
```


Check levels of temperature and condition (infection status)
```{r}
levels(dds$day)
```

```{r}
levels(dds$condition)
```


```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```

```{r}
armdrop_day_size <- res
# The main plot
plot(armdrop_day_size$baseMean, armdrop_day_size$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="SSWD Arm Drop With Size and Experiment Day  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
armdrop_day_size.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(armdrop_day_size.sig$baseMean, armdrop_day_size.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```



Get list of those 122 DEGs and write out to analyses for DESeq2:
Wrote out 2023-12-26. 
```{r}
#write.table(armdrop_day_size.sig, "../analyses/08-deseq2/DEGlist_armdrop8_with_size_and_expday.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```

# `join` deg lists with library counts
load the DEGlist_armdrop8_with_size_and_expday.tab
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_armdrop8_with_size_and_expday.tab")
head(deglist)
```

`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
armdropnorownames <- rownames_to_column(armdrop, var = "gene")
head(armdropnorownames)
```

```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, armdropnorownames, by = "gene")
head(deglistcount)
```

Write out the file:
```{r}
#write.table(deglistcount, "../analyses/08-deseq2/DEGlist_armdrop8_with_size_and_expday_genecounts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

# MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data:
```{r}
degheatmap <- select(degheatmap, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(degheatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```
Shouldn't lose any rows because this list is of the 122 DEGs of the 8 libraries at arm drop, taking into account star size/age and experimental day 

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with healthy or sick:
Add column annotations for the samples denoting temperature treatment and infection status:
```{r}
sample_col <- data.frame(size = c("large", "large", "large", "small", "small", "large", "small", "large"))
row.names(sample_col) <-colnames(nozerodegheatmap)
sample_col
```
specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "cyan3", exposed = "coral3")
)
```


```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```



A **contrast** is a linear combination of estimated log2 fold changes. Can be used to test if differences between groups are equal to zero.         
```{r}
resultsNames(dds)
```

```{r}
resSize <- results(dds,
                          contrast = c("condition", "small",  "large"))
head(resSize)
```

```{r}
sizecontrast <- resSize
# The main plot
plot(sizecontrast$baseMean, sizecontrast$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
sizecontrast.sig <- resSize[!is.na(resSize$padj) & resSize$padj <= 0.05, ]
points(sizecontrast.sig$baseMean, sizecontrast.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```
```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(resSize[!is.na(resSize$padj) & resSize$padj <= 0.05, ])
```
519 DEGs influenced by size that are something something 
```{r}
summary(resSize)
```


# Compare n=5 Adults at Arm Drop to n=3 Juveniles at Arm Drop

```{r}
armdrop <- select(countmatrix, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(armdrop)
```

Create a dataframe to get DEGs based on difference in age/size
```{r}
colData <- data.frame(condition=factor(c("large", "large", "large", "small", "small", "large", "small", "large")), 
                             type=factor(rep("paired-end", 8)))
rownames(colData) <- colnames(armdrop)
dds <- DESeqDataSetFromMatrix(countData = armdrop,
                                     colData = colData, 
                                     design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
```{r}
armdrop_size <- res
# The main plot
plot(armdrop_size$baseMean, armdrop_size$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="SSWD Arm Drop With Size  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
armdrop_size.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(armdrop_size.sig$baseMean, armdrop_size.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

Get list of those 112 DEGs and write out to analyses for DESeq2:
Wrote out 2023-12-26. 
```{r}
#write.table(armdrop_size.sig, "../analyses/08-deseq2/DEGlist_armdrop8_with_size.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

# `join` deg lists with library counts
load the DEGlist_armdrop8_with_size_and_expday.tab
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_armdrop8_with_size.tab")
head(deglist)
```

`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
armdropnorownames <- rownames_to_column(armdrop, var = "gene")
head(armdropnorownames)
```

```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, armdropnorownames, by = "gene")
head(deglistcount)
```

Write out the file:
```{r}
#write.table(deglistcount, "../analyses/08-deseq2/DEGlist_armdrop8_with_size_genecounts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out 20231227

# MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data:
```{r}
degheatmap <- select(degheatmap, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(degheatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```
Shouldn't lose any rows because this list is of the 112 DEGs of the 8 libraries at arm drop, taking into account star size/age and experimental day 

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with large or small:
Add column annotations for the samples denoting temperature treatment and infection status:
```{r}
sample_col <- data.frame(size = c("large", "large", "large", "small", "small", "large", "small", "large"))
row.names(sample_col) <-colnames(nozerodegheatmap)
sample_col
```
specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "cyan3", exposed = "coral3")
)
```


```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```


# For SICB: Compare Control vs Arm Drop (only looking at libraries from Day 8-15) 

**Clear environment**
Re-read in the countmatrix and round the integers. 


coelom_ID	star_size	experiment_day	disease_sign	star_ID
PSC.0149	large	                8	first_arm_drop	28
PSC.0150	large	                8	first_arm_drop	6
PSC.0174	large	                10	first_arm_drop	30
PSC.0177	small	                11	healthy	34
PSC.0186	large	                11	healthy	15
PSC.0187	small	                11	first_arm_drop	57
PSC.0188	small	                11	first_arm_drop	38
PSC.0190	large           	11	first_arm_drop	10
PSC.0202	large           	12	healthy	4
PSC.0203	large           	12	healthy	31
PSC.0209	large           	13	healthy	9
PSC.0219	small           	14	healthy	63
PSC.0228	small	                14	first_arm_drop	61
PSC.0230	small	                15	healthy	65
PSC.0231	large	                15	first_arm_drop	13

Subset the above libraries:
```{r}
#control (con) vs armdrop (ad)
convad <- select(countmatrix, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0177", "PSC.0186", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0202", "PSC.0203", "PSC.0209", "PSC.0219", "PSC.0228", "PSC.0230", "PSC.0231")
head(convad)
```

Get DEGs comparing the two groups - control vs first arm drop. 

```{r}
colData <- data.frame(condition=factor(c("firstarmdrop", "firstarmdrop", "firstarmdrop", "healthy", "healthy", "firstarmdrop", "firstarmdrop", "firstarmdrop", "healthy", "healthy", "healthy", "healthy", "firstarmdrop", "healthy", "firstarmdrop")), 
                             type=factor(rep("paired-end", 15)))
rownames(colData) <- colnames(convad)
dds <- DESeqDataSetFromMatrix(countData = convad,
                                     colData = colData, 
                                     design = ~ condition)
```
```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
```{r}
control_v_armdrop_fig <- res
# The main plot
plot(control_v_armdrop_fig$baseMean, control_v_armdrop_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="SSWD Exposure Status Healthy (Control) vs Exposed (Arm Drop) (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
control_v_armdrop_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(control_v_armdrop_fig.sig$baseMean, control_v_armdrop_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

Write out list of those 6441 DEGs:
Wrote out 2023-12-27. 
```{r}
#write.table(control_v_armdrop_fig.sig, "../analyses/08-deseq2/DEGlist_control_v_armdrop.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```

# `join` deg lists with library counts
load the `../analyses/08-deseq2/DEGlist_control_v_armdrop.tab`
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_control_v_armdrop.tab")
head(deglist)
```

`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
convadrownames <- rownames_to_column(convad, var = "gene")
head(convadrownames)
```

```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, convadrownames, by = "gene")
head(deglistcount)
```

Write out the file:
```{r}
#write.table(deglistcount, "../analyses/08-deseq2/DEGlist_control_v_armdrop_genecounts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

## MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data:    
Pull the columns out in order where controls are first:     
```{r}
degheatmap <- select(degheatmap, "PSC.0177", "PSC.0186", "PSC.0202", "PSC.0203", "PSC.0209", "PSC.0219", "PSC.0230", "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0187", "PSC.0188", "PSC.0190", "PSC.0228", "PSC.0231")
head(degheatmap)
```

Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```
Shouldn't lose any rows because this list is of the 6441 DEGs of the 8 libraries at arm drop, taking into account star size/age and experimental day 

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with healthy or sick:
Add column annotations for the samples denoting temperature treatment and infection status:
```{r}
sample_col <- data.frame(diseasesign = c("control", "control", "control", "control", "control", "control", "control", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop"))
row.names(sample_col) <-colnames(nozerodegheatmap)
sample_col
```
add annotation for star size:
```{r}
sample_col$star.size <- c("small", "large", "large", "large", "large", "small", "small", "large", "large", "large", "small", "small", "large", "small", "large")
sample_col
```



specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "cyan3", exposed = "coral3"),
  star.size = c(large = "#AF2CEC", small = "#2C75EC")
)
```


```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```

annotate the DEGlist with the BLASToutput, too:
Read in the DEGlist:
```{r}
deglistcount <- read.delim("../analyses/08-deseq2/DEGlist_control_v_armdrop_genecounts.tab")
head(deglistcount)
```

read in blast output of the gene list from the genome and uniprot:
```{r}
blastout <- read.delim("../analyses/10-BLAST/summer2022-uniprot_blastx_sep.tab", header = FALSE)
head(blastout)
```

rename the first column "gene":
```{r}
cols <- c("gene", "V2", "uniprotID", "geneID", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
colnames(blastout) <- cols
head(blastout)
```

`join` the blastout with the deglistcount:
```{r}
deglistcountblast <- left_join(deglistcount, blastout, by = "gene")
head(deglistcountblast)
```

write out the blast annotated file:
```{r}
#write.table(deglistcountblast, "../analyses/08-deseq2/DEGlist_control_v_armdrop_genecounts_blast.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

# compare first vs second arm drop:
coelom_ID	star_size	experiment_day	disease_sign	star_ID
PSC.0149	large	                8	first_arm_drop	28
PSC.0150	large           	8	first_arm_drop	6
PSC.0174	large           	10	first_arm_drop	30
PSC.0188	small           	11	first_arm_drop	38
PSC.0228	small           	14	first_arm_drop	61
PSC.0156	large           	9	second_arm_drop	28
PSC.0198	small           	12	second_arm_drop	38
PSC.0206	large           	12	second_arm_drop	30
PSC.0217	large           	13	second_arm_drop	6
PSC.0235	small           	15	second_arm_drop	61

```{r}
fvsad <- select(countmatrix, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0188", "PSC.0228", "PSC.0156", "PSC.0198", "PSC.0206", "PSC.0217", "PSC.0235")
head(fvsad)
```

Create a dataframe to get DEGs based on FIRST VS SECOND ARM DROP: 
```{r}
colData <- data.frame(condition=factor(c("firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop")), 
                             type=factor(rep("paired-end", 10)))
rownames(colData) <- colnames(fvsad)
dds <- DESeqDataSetFromMatrix(countData = fvsad,
                                     colData = colData, 
                                     design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
only 3 DEGs??? 

```{r}
firstvssecondarmdrop <- res
# The main plot
plot(firstvssecondarmdrop$baseMean, firstvssecondarmdrop$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="First Arm Drop VS Second Arm Drop (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
firstvssecondarmdrop.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(firstvssecondarmdrop.sig$baseMean, firstvssecondarmdrop.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

Get list of those 3 DEGs and write out to analyses for DESeq2:
Wrote out 2023-12-27. 
```{r}
#write.table(firstvssecondarmdrop.sig, "../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```

# `join` deg lists with library counts
load the ../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop.tab
```{r}
deglist <- read.delim("../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop.tab")
head(deglist)
```
`join` with the count data from the libraries i compared:
make row names for both files into a column called "gene"
```{r}
fvsrownames <- rownames_to_column(fvsad, var = "gene")
head(fvsrownames)
```

```{r}
deglist <- rownames_to_column(deglist, var = "gene")
head(deglist)
```

```{r}
deglistcount <- left_join(deglist, fvsrownames, by = "gene")
head(deglistcount)
```

Write out the file:
```{r}
#write.table(deglistcount, "../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop_genecounts.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

`join` `../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop_genecounts.tab` with blastoutput:

```{r}
deglistcount <- read.delim("../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop_genecounts.tab")
head(deglistcount)
```
```{r}
blastout <- read.delim("../analyses/10-BLAST/summer2022-uniprot_blastx_sep.tab", header = FALSE)
head(blastout)
```

rename the first column "gene":
```{r}
cols <- c("gene", "V2", "uniprotID", "geneID", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
colnames(blastout) <- cols
head(blastout)
```

`join` blastout with deglistcount by gene column:
`join` the blastout with the deglistcount:
```{r}
deglistcountblast <- left_join(deglistcount, blastout, by = "gene")
head(deglistcountblast)
```
write out table:
write out the blast annotated file:
```{r}
#write.table(deglistcountblast, "../analyses/08-deseq2/DEGlist_firstarmdrop_vs_secondarmdrop_genecounts_blast.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
wrote out 20231227

# MAKE HEATMAP
Load in packages:
```{r}
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```

Set heatmap colors:
```{r}
heatmapPurpleOrange <- RColorBrewer::brewer.pal(11, "PuOr")
```

Set `gene` column as rownames:
```{r}
rownames(deglistcount) <- deglistcount$gene #set gene IDs as rownames
degheatmap <- deglistcount[,-1] #remove redundant column
head(degheatmap)
```

Pull out just the sample count data:
```{r}
degheatmap <- select(degheatmap, "PSC.0149", "PSC.0150", "PSC.0174", "PSC.0188", "PSC.0228", "PSC.0156", "PSC.0198", "PSC.0206", "PSC.0217", "PSC.0235")
head(degheatmap)
```
Going to remove all rows that have "0" for all samples: 
```{r}
nozerodegheatmap <- degheatmap[rowSums(degheatmap[, -1] >0) !=0, ]
head(nozerodegheatmap)
```
Shouldn't lose any rows because this list is of the 3 DEGs of the 10 libraries at arm drop, taking into account star size/age and experimental day 

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 15, fontsize_row = 12)
```

Annotate heatmap with first arm or second arm drop:
Add column annotations for the samples denoting first or second arm drop status:
```{r}
sample_col <- data.frame(armdrop = c("firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "firstarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop", "secondarmdrop"))
row.names(sample_col) <-colnames(nozerodegheatmap)
sample_col
```

specify colors for annotation columns:
```{r}
my_colors <- list(
  treatment = c(control = "#1DB6AB", exposed = "#C23F0E"),
  star.size = c(large = "#AF2CEC", small = "#82ed82"),
  experiment.day = c(day8 = "#ECA02C", day10 = "#ECE32C", day11 = "#66EC2C", day14 = "#2C75EC", day9 = "#F2A2EC", day12 = "#C79BE8", day13 = "#56525A", day15 = "#A39FA7")
)
```

```{r}
pheatmap(nozerodegheatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapPurpleOrange, fontsize_col = 12, fontsize_row = 12)
```


Annotate columns with star size and experiment day:
add star size:
```{r}
sample_col$star.size <- c("large", "large", "large", "small", "small", "large", "small", "large", "large", "small")
sample_col
```

add eximperiment day info:
```{r}
sample_col$experiment.day <- c("day8", "day8", "day10", "day11", "day14", "day9", "day12", "day12", "day13", "day15")
sample_col
```
