---
title: "10-BLAST-summ22_phelgenome"
output: html_document
date: "2023-12-22"
---
Rmd to perform BLAST with the _Pycnopodia helianthoides_ genome gene list against the published _Pycnopodia helianthoides_ genome. 

Genome gene list: `project-pycno-sizeclass-2022/data/augustus.hints.codingseq`

Based on this jupyter notebook by Steven Roberts: https://github.com/RobertsLab/code/blob/master/09-blast.ipynb 

```{bash}
pwd
```

```{bash}
/home/shared/ncbi-blast-2.15.0+/bin/blastx -h
```


# Create a BLAST database
I would like to make a database of UniProt/Swiss-prot. see https://www.uniprot.org/downloads 

```{bash}
cd /home/shared/8TB_HDD_02/graceac9/GitHub/project-pycno-sizeclass-2022/analyses/10-BLAST
curl -O https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
mv uniprot_sprot.fasta.gz uniprot_sprot_r2021_03.fasta.gz
gunzip -k uniprot_sprot_r2021_03.fasta.gz
cd -
```

```{bash}
pwd
```

note: couldn't figure out syntax errors in code chunk below, so I used `rsync` to transfer the database directory I made locally into `/home/shared/8TB_HDD_02/graceac9/GitHub/project-pycno-sizeclass-2022/analyses/10-BLAST`
```{python}
#!/home/shared/ncbi-blast-2.15.0+/bin/makeblastdb \
#-in ../analyses/10-BLAST/uniprot_sprot_r2021_03.fasta \
#-dbtype prot \
#-out ../analyses/10-BLAST/analyses/10-BLAST/uniprot_sprot_r2021_03
```

# Get a Query Sequence
```{bash}
pwd
```

`rsync` the data/augustus.hints.codingseq to the raven code directory. 


```{python}
#how many sequences? lets count ">" as we know each contig has 1
!grep -c ">" augustus.hints.codingseq
```

# Run BLAST

```{bash}
pwd
```


```{python}
!/home/shared/ncbi-blast-2.15.0+/bin/blastx \
-query augustus.hints.codingseq \
-db ../analyses/10-BLAST/blast/uniprot_sprot_r2021_03 \
-out ../analyses/10-BLAST/summer2022-uniprot_blastx.tab \
-evalue 1E-20 \
-num_threads 8 \
-max_target_seqs 1 \
-outfmt 6
```



